src/ts/contentScripts/duplicate-block-content/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { Feature } from '../../utils/settings'
+import { Roam, RoamNode, Selection } from '../../utils/roam';
+
+export const config: Feature = {
+    id: 'duplicate',
+    name: 'Duplicate',
+    shortcuts: [
+        {
+            id: 'dupShortcut', label: 'Shortcut for duplication', initValue: '',
+            onPress: () => duplicate()
+        },
+    ]
+}
+
+const duplicate = () => {
+    const node = Roam.getActiveRoamNode();
+    const selectedText = node?.selectedText();
+
+    if (node && selectedText) {
```

Comment:
checking for node is redundant here.

Reply:
TS: Object is possibly 'null'.
how you get rid of that in 3 places below?

Reply:
ah, ts not so smart in this regard =\

the following can be one alternative, I guess either is fined though
```
    if (Roam.getActiveRoamNode()?.selectedText()) {

        Roam.applyToCurrent(node => {
            const selectionLength = node.selectedText().length;
            const selectionEnd = node.selection.end
            const newText = node.text.substring(0, selectionEnd)
                + node.selectedText()
                + node.text.substring(selectionEnd);

            return new RoamNode(newText, new Selection(selectionEnd, selectionEnd + selectionLength))
        })

```